File: examples/python/consultar_rut.py
"""Consulta una empresa chilena por RUT.

Uso:
    API_KEY=tu_api_key python consultar_rut.py 93834000-5

Requiere: pip install requests
"""
import os
import sys

import requests

API_BASE = "https://api-sii-chile.webempresario.com/v1"


def consultar_empresa_por_rut(rut: str, api_key: str) -> dict:
    resp = requests.get(f"{API_BASE}/{rut}", headers={"X-Api-Key": api_key}, timeout=10)
    resp.raise_for_status()
    return resp.json()


if __name__ == "__main__":
    if len(sys.argv) < 2:
        sys.exit("Uso: API_KEY=tu_api_key python consultar_rut.py <rut-con-dv>")

    rut_arg = sys.argv[1]
    api_key_env = os.environ.get("API_KEY")
    if not api_key_env:
        sys.exit("Define la variable de entorno API_KEY con tu clave")

    empresa = consultar_empresa_por_rut(rut_arg, api_key_env)
    print(empresa)

    # Ejemplo: acceder a las coordenadas del primer domicilio
    domicilios = empresa.get("data", {}).get("domicilios", [])
    if domicilios and domicilios[0].get("GEO", {}).get("LAT"):
        geo = domicilios[0]["GEO"]
        print(f"\nCoordenadas del domicilio principal: {geo['LAT']}, {geo['LON']}")

File: examples/javascript/consultar_rut.js
// Consulta una empresa chilena por RUT.
// Funciona en Node.js 18+ (fetch nativo) y en el navegador.
//
// Uso:
//   API_KEY=tu_api_key node consultar_rut.js 93834000-5

async function consultarEmpresaPorRut(rut, apiKey) {
  const res = await fetch(`https://api-sii-chile.webempresario.com/v1/${rut}`, {
    headers: { 'X-Api-Key': apiKey },
  });

  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(`API error ${res.status}: ${err.message || res.statusText}`);
  }

  return res.json();
}

// --- Ejemplo de uso (Node.js) ---
if (typeof require !== 'undefined' && require.main === module) {
  const rut = process.argv[2];
  const apiKey = process.env.API_KEY;

  if (!rut || !apiKey) {
    console.error('Uso: API_KEY=tu_api_key node consultar_rut.js <rut-con-dv>');
    process.exit(1);
  }

  consultarEmpresaPorRut(rut, apiKey)
    .then((data) => {
      console.log(JSON.stringify(data, null, 2));

      // Ejemplo: acceder a las coordenadas del primer domicilio
      const primerDomicilio = data.data.domicilios[0];
      if (primerDomicilio?.GEO?.LAT) {
        console.log(`\nCoordenadas del domicilio principal: ${primerDomicilio.GEO.LAT}, ${primerDomicilio.GEO.LON}`);
      }
    })
    .catch((err) => {
      console.error(err.message);
      process.exit(1);
    });
}

module.exports = { consultarEmpresaPorRut };
